projects/Mallard/src/components/weather.tsx

```diff
@@ -286,20 +294,36 @@ const WeatherWithForecast = ({ weather }: { weather: Weather }) => {
             </View>
         )
     }
-
-    // FIXME: We really should validate data after fetching, not during
-    // rendering. That way the error would get handled further up the chain
-    // instead of rendering a blank space with no logging.
     return <></>
 }
 
-const WeatherWidget = React.memo(() => {
-    const query = useQuery<QueryData>(QUERY)
-    if (query.loading) return null
+const WeatherWidget = React.memo(
+    ({ isWeatherShown }: { isWeatherShown: boolean }) => {
+        if (!isWeatherShown) {
+            return <View style={styles.emptyWeatherSpace} />
+        }
+
+        const query = useQuery<QueryData>(QUERY)
```

Comment:
can't call a hook conditionally, this will crash React

Reply:
Interesting. How is it conditional?

Reply:
Simply by the fact that React.memo only runs if the prop passed to it changes?

Reply:
useQuery itself isn't inside an if {} block.